### ISSUE TITLE
Ensure #pragma once is at top of files.

### ISSUE THREAD

**Issue by elliotcmorris** [OpenAssetIO/OpenAssetIO#1103]
## What
Ensure `#pragma once` is always at the top of the file.
## Why
In some of our files `#pragma once` comes after some `#include` declarations. For example the files in the `log` directory. [Example.](https://github.com/OpenAssetIO/OpenAssetIO/blob/bef5b5cd165601a94faecbcf00dc3d840a90fd4b/src/openassetio-core/include/openassetio/log/ConsoleLogger.hpp#L7)

`#pragma once` is subject to macro expansions, and it's conceivable, albeit unlikely, that this could cause a hard to find issue.

If we ever get a moment we should just move these up. Definately a minor odd-job.

## Acceptance Criteria
`#pragma once` is always at the very top of all cpp headers files in the repo.

**Comment by Gmaj9** [comment]
Hello, I would like to take this on for ASWF Dev Days.

**Comment by Gmaj9** [comment]
I only found one .h/.hpp file that didn't have `#pragma once` at the top. However, I noticed whitespace inconsistencies throughout. In some places `#pragma once` is directly beneath the comment at the top of the file; other times there is a newline in between the two. The same goes for whether or not there is a newline between `#pragma once` and the `#include` statements following it. Should I address this, and if so, what would be preferred?

**Comment by feltech** [comment]
> I noticed whitespace inconsistencies throughout.

I like the way you think! Personally, I'd love to be militant about consistency. But realistically, we have to draw the line somewhere, for pragmatic reasons 😉 .  So I'd say, if you want to make them consistent, go for it, I'll happily review it. But it's really not necessary.

In terms of preference, after looking at a random selection of headers, I think the most common case is
```
// Copyright stuff
/**
 * Optional (rare) file docstring
 */
#pragma once

#include <blah> 
```

So converging on that most-common style would be best.

**Comment by Gmaj9** [comment]
OK. And — if I read the docs correctly — the whitespace changes can go in this PR, but each in their own commit?

**Comment by feltech** [comment]
> OK. And — if I read the docs correctly — the whitespace changes can go in this PR, but each in their own commit?

Yeah separate commits would be good. But since the two sets of changes are so closely related it could also make sense as a single commit.

**Comment by Gmaj9** [comment]
Apologies for the delay. Should I prefix my commit titles with `[Lint]`? It looks appropriate although I see no component used for the like commit https://github.com/OpenAssetIO/OpenAssetIO/commit/d53a792c9c81692e847fb2dc849260b2ca826062.

**Comment by feltech** [comment]
> Apologies for the delay. Should I prefix my commit titles with `[Lint]`? It looks appropriate although I see no component used for the like commit [d53a792](https://github.com/OpenAssetIO/OpenAssetIO/commit/d53a792c9c81692e847fb2dc849260b2ca826062).

I think, since this is technically a bug fix (albeit very minor) then `[Core]` would be appropriate.  Elliot was a bit naughty forgetting the component tag in that previous commit (and I was naughty not spotting it in review!).
